Lib/posixpath.py

```diff
@@ -572,7 +570,7 @@ def commonpath(paths):
                 common = s1[:i]
                 break
 
-        prefix = sep if isabs else sep[:0]
+        prefix = sep if paths[0][:1] == sep else sep[:0]
```

Comment:
Why not keep `startswith`? Can you also benchmark withs strings that have lengths > 1? The length 1 string '/' you used might have some special behavior.

Reply:
I can't benchmark `startswith()` that on 3.12.

Reply:
Actually, I can just replace the slice comparison in the reference code with `startwith()`. I'll re-run the benchmark with `foo` soon.